### PR TITLE
Implement sessions search endpoint

### PR DESCRIPTION
**Pull request by dvrylc**
Sample response object: 

```json
{
  "sessions": {
    "alice.b.tmms@gmail.tmt": {
      "openSessions": {
        "Second team feedback session [Sun, 01 Apr 2012, 11:59 PM SGT - Thu, 02 Apr 2037, 11:59 PM SGT]": "http://localhost:4200/web/sessions/submission?courseid=teammates.instructor.uni-demo&fsname=Second+team+feedback+session&key=EBE50B16D49CF8511F63AC64F55EEE9A50E093BE541FA78D17D6198DCFF6F686206709FBDC0E43EC742E25403B26318ABE1ACDFA61D628BD1931338B345275CC&studentemail=alice.b.tmms%40gmail.tmt"
      },
      "closedSessions": {
        "Session with different question types [Sun, 01 Apr 2012, 11:59 PM SGT - Mon, 02 Apr 2012, 11:59 PM SGT] (Currently Not Open)": "http://localhost:4200/web/sessions/submission?courseid=teammates.instructor.uni-demo&fsname=Session+with+different+question+types&key=EBE50B16D49CF8511F63AC64F55EEE9A50E093BE541FA78D17D6198DCFF6F686206709FBDC0E43EC742E25403B26318ABE1ACDFA61D628BD1931338B345275CC&studentemail=alice.b.tmms%40gmail.tmt",
        "First team feedback session [Sun, 01 Apr 2012, 11:59 PM SGT - Mon, 02 Apr 2012, 11:59 PM SGT] (Currently Not Open)": "http://localhost:4200/web/sessions/submission?courseid=teammates.instructor.uni-demo&fsname=First+team+feedback+session&key=EBE50B16D49CF8511F63AC64F55EEE9A50E093BE541FA78D17D6198DCFF6F686206709FBDC0E43EC742E25403B26318ABE1ACDFA61D628BD1931338B345275CC&studentemail=alice.b.tmms%40gmail.tmt"
      },
      "publishedSessions": {
        "First team feedback session [Sun, 01 Apr 2012, 11:59 PM SGT - Mon, 02 Apr 2012, 11:59 PM SGT] (Published)": "http://localhost:4200/web/sessions/result?courseid=teammates.instructor.uni-demo&fsname=First+team+feedback+session&key=EBE50B16D49CF8511F63AC64F55EEE9A50E093BE541FA78D17D6198DCFF6F686206709FBDC0E43EC742E25403B26318ABE1ACDFA61D628BD1931338B345275CC&studentemail=alice.b.tmms%40gmail.tmt",
        "Session with different question types [Sun, 01 Apr 2012, 11:59 PM SGT - Mon, 02 Apr 2012, 11:59 PM SGT] (Published)": "http://localhost:4200/web/sessions/result?courseid=teammates.instructor.uni-demo&fsname=Session+with+different+question+types&key=EBE50B16D49CF8511F63AC64F55EEE9A50E093BE541FA78D17D6198DCFF6F686206709FBDC0E43EC742E25403B26318ABE1ACDFA61D628BD1931338B345275CC&studentemail=alice.b.tmms%40gmail.tmt"
      }
    }
  },
  "requestId": "5e5203000007c830511a418e"
}
```

Under the `sessions` object, the keys are currently the student's email address but this is flexible - we can choose any of the fields in `StudentAttributes`.

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/action/SearchSessionsAction.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        // Only admins can search for sessions
+        if (!userInfo.isAdmin) {
+            throw new UnauthorizedAccessException("Admin privilege is required to access this resource.");
+        }
+    }
+
+    @Override
+    public ActionResult execute() {
+        String searchKey = getNonNullRequestParamValue(Const.ParamsNames.ADMIN_SEARCH_KEY);
+        List<StudentAttributes> students = logic.searchStudentsInWholeSystem(searchKey).studentList;
```
Comment (hcwong) [inline src/main/java/teammates/ui/webapi/action/SearchSessionsAction.java:47]: Is this endpoint also called for instructor search? if so, we would also have to search for instructors and maybe return a {students: "link for students" and instructors: "links for instructors"}

Comment (dvrylc) [reply src/main/java/teammates/ui/webapi/action/SearchSessionsAction.java:47]: As far as I can see, session links are only returned for the admin search: https://github.com/dvrylc/teammates/pull/1#issuecomment-589702428

**Review by hcwong — APPROVED** [review]
I'm getting an error when running `generateTypes`, but this could be a local issue. Could you check if this happens on your end.
The error i get is `Warning: Multiple classes are mapped to 'Instructor' name. Conflicting classes: [class teammates.ui.webapi.output.InstructorData, class teammates.storage.entity.Instructor]`
Don't think the error is related to the changes though, so I'm approving first


Leaving this here for easy reference: notOpenSessions -> closedSessions

**Comment by dvrylc** [comment]
> I'm getting an error when running `generateTypes`, but this could be a local issue. Could you check if this happens on your end.

Oof good catch! Had to rebase, not sure how the build even passed previously 🤔 can you verify?

**Comment by hcwong** [comment]
> > I'm getting an error when running `generateTypes`, but this could be a local issue. Could you check if this happens on your end.
> 
> Oof good catch! Had to rebase, not sure how the build even passed previously 🤔 can you verify?

Hmm yea works after rebase, merge merge merge. on a side note, we should totally run teammates in a container. running it on local always gives surprises

**Pull request merged**